### ISSUE TITLE
solve_ivp fails with np.errstate(all='raise')

### ISSUE THREAD

**Issue by slayoo** [scipy/scipy#11064]
#### Reproducing code example (based on the example from the docs):
```python
from scipy.integrate import solve_ivp
import numpy as np

def exponential_decay(t, y): return -0.5 * y
with np.errstate(all='raise'):
    sol = solve_ivp(exponential_decay, [0, 10], [2, 4, 8])
```

#### Error message:
```
---------------------------------------------------------------------------
FloatingPointError                        Traceback (most recent call last)
<ipython-input-2-c042dd67a7ba> in <module>()
      4 def exponential_decay(t, y): return -0.5 * y
      5 with np.errstate(all='raise'):
----> 6     sol = solve_ivp(exponential_decay, [0, 10], [2, 4, 8])

/usr/lib/python3/dist-packages/scipy/integrate/_ivp/ivp.py in solve_ivp(fun, t_span, y0, method, t_eval, dense_output, events, vectorized, **options)
    473     status = None
    474     while status is None:
--> 475         message = solver.step()
    476 
    477         if solver.status == 'finished':

/usr/lib/python3/dist-packages/scipy/integrate/_ivp/base.py in step(self)
    180         else:
    181             t = self.t
--> 182             success, message = self._step_impl()
    183 
    184             if not success:

/usr/lib/python3/dist-packages/scipy/integrate/_ivp/rk.py in _step_impl(self)
    111         atol = self.atol
    112 
--> 113         min_step = 10 * np.abs(np.nextafter(t, self.direction * np.inf) - t)
    114 
    115         if self.h_abs > max_step:

FloatingPointError: underflow encountered in nextafter
```

#### Scipy/Numpy/Python version information:
```python
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
```
```
1.1.0 1.16.1 sys.version_info(major=3, minor=7, micro=2, releaselevel='final', serial=0)
```

**Comment by drhagen** [comment]
Presumably, this kind of thing can happen with lots of SciPy algorithms. What is the standard SciPy response? Does SciPy mark whole algorithms with `ignore`, just particular lines where signals are most likely, or is this expected and desired behavior when setting `raise` on everything?

**Comment by mdhaber** [comment]
TLDR @seberg : outside of a `np.errstate(all='warn')` or similar context, is it normal for overflows to emit a visible warning but not underflows?
```python3
import numpy as np
np.float64(1e-300) / 1e300  # no warning
np.float64(1)/ 0  # RuntimeWarning: divide by zero encountered in double_scalars
```

---
IIUC, the purpose of the context `np.errstate(all='raise')` is to raise an error when any one of a certain class of warnings occurs, so it is not surprising to me that when such a warning occurs within the context, an error is raised.

Perhaps what is surprising is that without any error context, some classes of warnings are emitted, e.g.:
```python3
import numpy as np
np.float64(1)/ 0  # RuntimeWarning: divide by zero encountered in double_scalars
```
whereas this class of warning is not emitted:
```python3
import numpy as np
np.float64(1e-300) / 1e300
```
yet within the `np.errstate(all='raise')` context, it still raises:
```python3
import numpy as np
with np.errstate(all='raise'):  # under='raise', specifically
    np.float64(1e-300) / 1e300  # FloatingPointError: underflow encountered in double_scalars
```

In the context of this issue, it happens here:
https://github.com/scipy/scipy/blob/f5f64c55899e3b3d3b27f9566d310c434401a8a7/scipy/integrate/_ivp/rk.py#L119
```python3
import numpy as np
np.nextafter(0, -1 * np.inf)  # -5e-324, a subnormal number, no warning emitted
with np.errstate(under='raise'):
    np.nextafter(0, -np.inf)  # FloatingPointError: underflow encountered in nextafter
```

Whatever is going on, it doesn't seem like a SciPy issue. 
I'm not sure whether this is intended behavior, so I don't plan to file an issue with NumPy. Please feel free to do so, or re-open this issue (stating what the expected behavior is) if I've missed the point.

**Comment by seberg** [comment]
Underflows are off by default, and I suspect that is for the best.  Underflows are not typically interesting.  Maybe "all" is a bit annoying in that sense, since chances are most users don't care about underflows.

**Comment by mdhaber** [comment]
Thanks @seberg!

**Comment by mdhaber** [comment]
On the other hand, maybe having a subnormal minimum step is not good. I'll go ahead and change this a bit so that the minimum step is the smallest normal float. Maybe it has something to do with gh-11239.
